### PR TITLE
Update bazel_to_cmake to support testonly for iree_bytecode_module

### PR DESCRIPTION
**Pull request by not-jenni**
Also adds the testonly option to the iree_bytecode_modules that are only used for testing

### CODE REVIEW

**Review by GMNGeoffrey — APPROVED** [review]
Please run buildifier

**Comment by not-jenni** [comment]
Oops looks like testonly is only supported for CMake for iree_bytecode_module, I will look into adding it for bazel

**Comment by GMNGeoffrey** [comment]
> Oops looks like testonly is only supported for CMake for iree_bytecode_module, I will look into adding it for bazel

Oh it should be forwarding kwargs to the underlying genrule, I think

File: iree/tools/compilation.bzl [hunk 1]
```diff
@@ -23,7 +23,9 @@ def iree_bytecode_module(
         flags = ["-iree-mlir-to-vm-bytecode-module"],
         translate_tool = "//iree/tools:iree-translate",
         cc_namespace = None,
-        visibility = None):
+        visibility = None,
+        testonly = False,
+        **kwargs):
```
Comment (GMNGeoffrey) [inline iree/tools/compilation.bzl:28]: If you specify the kwargs, you shouldn't need add these manually. You should probably also forward the kwargs to both the underlying rules (implicitly creating cc_embed_data is one of the things I hate about this macro. See https://docs.bazel.build/versions/master/skylark/bzl-style.html#macros). And it's weird that visibility here doesn't get passed to the genrule too. The only thing that need to get pulled out of kwargs are args that don't make sense to just forward to both rules. Generally you'd expect a macro to take all the standard kwargs defined in all rules.

Comment (not-jenni) [reply iree/tools/compilation.bzl]: Fixed!

**Review by GMNGeoffrey — APPROVED** [review]

**Pull request merged**